sdks/python/apache_beam/transforms/core.py

```diff
@@ -1975,10 +1990,14 @@ def add_input_types(transform):
       return combined
 
     if self.has_defaults:
-      combine_fn = (
-          self.fn if isinstance(self.fn, CombineFn) else
-          CombineFn.from_callable(self.fn))
-      default_value = combine_fn.apply([], *self.args, **self.kwargs)
+      combine_fn = copy.deepcopy(
+          self.fn if isinstance(self.fn, CombineFn) else CombineFn.
```

Comment:
nit: this can be `copy.deepcopy(self.fn) if...` i.e. copy is only needed in the first branch